packages/components/src/components/hds/advanced-table/helpers.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import type { HdsAdvancedTableTdSignature } from './td';
+import type { HdsAdvancedTableThSignature } from './th';
+
+const getAllFocusableChildren = (element: HTMLElement): NodeListOf<Element> => {
+  return element.querySelectorAll(
+    'button:not([disabled]), a[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]:not([tabindex="-1"]):not([disabled])'
```

Comment:
noting here that this list is not exhaustive, but rather elements we expect to find in a table

Reply:
Yeah, I was actually going to ask about this. I know we use ember-focus-trap, which is dependent on [focus-trap](https://github.com/focus-trap/focus-trap?tab=readme-ov-file) and [tabbable](https://github.com/focus-trap/tabbable).

I can't use ember-focus trap in this implementation, but would it make sense to use normal focus-trap and tabbable to be more confident in the focus trapping within the cell and finding all focusable children?

Reply:
I believe it's worth exploring it, considering it may reduce the code we need to maintain, and we use `ember-focus-trap` elsewhere, so it's not increasing our bundle size.

Reply:
It may also solve the issue flagged above as 'keyboard navigation + expandable rows'

Reply:
Update: I refactored so the cells use ember-focus-trap and the helpers functions use tabbable now! Also, the commit fixes the issues with arrow keys breaking the nav when youre inside of a cell (tldr: the arrow keys should only work when focused on a cell, so now you can only tab between elements inside of a cell)